dipy/reconst/dti.py

```diff
@@ -777,17 +782,35 @@ def fit(self, data, mask=None):
             min_signal = self.min_signal
 
         data_in_mask = np.maximum(data_in_mask, min_signal)
-        params_in_mask = self.fit_method(self.design_matrix, data_in_mask,
-                                         *self.args, **self.kwargs)
+
+        if self.return_S0_hat:
```

Comment:
I agree on the input side that's fine but the output of fit_method is a little more tricky because they either return 1 or 2 arguments. So I think the alternative below still requires an if statement but is perhaps more clean?

```python
params_in_mask = self.fit_method(
                self.design_matrix,
                data_in_mask,
                return_S0_hat=self.return_S0_hat,
                *self.args,
                **self.kwargs)
if self.return_S0_hat:
    params_in_mask, model_S0 = params_in_mask
```